### PR TITLE
Testing / Moving part

### PR DESCRIPTION
**Pull request by The0mikkel**

### CODE REVIEW

File: Common/src/test/java/dk/sdu/sesem4/common/data/EntityParts/MovingPartTest.java [hunk 1]
```diff
@@ -119,4 +120,47 @@ public void testDirectionWhenNoMovementControllerShouldBeNull() {
...
+		for (Direction direction : Direction.values()) {
+			when(movementControllerSPI.getMovement(gameData, entity)).thenReturn(direction);
+
+			mockedMovingPart.process(gameData, entity);
+
+			verify(positionPart).setDirection(direction);
+		}
+
+		// Verify if no direction is ot be returned from movement controller
+		when(movementControllerSPI.getMovement(gameData, entity)).thenReturn(null);
```
Comment (Muhammed-als) [inline Common/src/test/java/dk/sdu/sesem4/common/data/EntityParts/MovingPartTest.java:154]: Are we using Direction.None or null?

**Review by Muhammed-als — COMMENTED** [review]
LGTM, but I dont really understand if we are going to use None direction or null?

File: Common/src/test/java/dk/sdu/sesem4/common/data/EntityParts/MovingPartTest.java (hunk 1 above)
Comment (The0mikkel) [reply Common/src/test/java/dk/sdu/sesem4/common/data/EntityParts/MovingPartTest.java:154]: We would like to not use the `Direction.NONE`, as the system is build around the null being the none direction. `null` represents nothing, and having a "direction" state being "nothing" is weird, so better to use `null` for that, in my opinion.

**Review by JfriisJ — APPROVED** [review]
OK

**Review by JfriisJ — APPROVED** [review]
LGTM

**Review by MGertz — APPROVED** [review]
LGTM

**Review by JfriisJ — APPROVED** [review]
OK

**Review by Muhammed-als — APPROVED** [review]
LGTM

**Pull request merged**